grails-app/controllers/au/org/emii/aatams/SensorController.groovy

```diff
@@ -32,8 +32,6 @@ class SensorController extends ReportController {
         def tag = tagFactoryService.lookupOrCreate(params.tag)
         assert(tag)
 
-        params.tag = tag
```

Comment:
PS.  I believe the reason why this fixes this particular problem is because we were adding an actual tag domain containing a device_model entity to parameters which included device_model changed details (id) causing hibernate to get confused and try to update the id of the device_model instead of changing the reference to be to another device_model.